sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -793,11 +793,6 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
    */
   override def visitInsertOverwriteDir(
       ctx: InsertOverwriteDirContext): InsertDirParams = withOrigin(ctx) {
-    if (ctx.LOCAL != null) {
```

Comment:
Seems `ctx.LOCAL != null` should be returned instead of `false`. Also, it seems "insert overwrite into local" with data sources of spark isn't implemented yet.

Reply:
but i see we can do
 `insert into overwrite directory file:/// using <datasource>` where the output is persisted in local filesystem. Can you please correct me how this is different than doing 
`insert into local directory overwrite file:/// using <datasource>`

Reply:
`insert into local directory overwrite /a/b/c <datasource>` will write into the local file system regardless of default file system; however, `insert into directory overwrite /a/b/c <datasource>` will write according to the default scheme (e.g., `fs.default.name`).

Reply:
Ok i see, so part of this PR i will enforce `<path>` scheme as `file:///` when its `insert overwrite local directory <path>` , that should make the command support. What do you suggest.?

Reply:
Yeah, I think that's why people didn't bother implement this so far. If the changes required to implement this is rather small, we can try. Otherwise, I wouldn't since it's there more for hive compatibility.

Reply:
thanks for inputs @HyukjinKwon, now i have updated the PR to make sure scheme is local when we use insert overwrite local directory. This should suffice. Please suggest.